extensions/spoons/init.lua

```diff
@@ -164,4 +206,152 @@ function module.list(only_loaded)
+      else
+         log.ef("Invalid JSON received for repository '%s': %s", repo, body)
+      end
+   end
+end
+
+--- hs.spoons.updateRepo(repo)
+--- Method
+--- Fetch and store locally the information about the contents of a Spoons repository
+---
+--- Parameters:
+---  * repo - name of the repository to update. Defaults to `"default"`.
+---
+--- Returns:
+---  * None
+function module.updateRepo(repo)
+   if not repo then repo = 'default' end
+   if module.repos[repo] and module.repos[repo].url then
+      module.repos[repo].json_url = string.gsub(module.repos[repo].url, "/$", "") .. "/raw/master/docs/docs.json"
+      module.repos[repo].download_base_url = string.gsub(module.repos[repo].url, "/$", "") .. "/raw/master/Spoons/"
+      hs.http.asyncGet(module.repos[repo].json_url, nil, hs.fnutils.partial(_storeRepoJSON, repo))
+   else
+      log.ef("Invalid or unknown repository '%s'", repo)
+   end
+end
+
+--- hs.spoons.updateAllRepos()
+--- Method
+--- Fetch and store locally the information about the contents of all registered Spoons repositories
+---
+--- Parameters:
+---  * None
+---
+--- Returns:
+---  * None
+function module.updateAllRepos()
+   for k,v in pairs(module.repos) do
+      module.updateRepo(k)
+   end
+end
+
+-- --------------------------------------------------------------------
+-- Spoon installation
+
+-- Internal callback function to finalize the installation of a spoon after the zip file has been downloaded.
+-- This is used by hs.spoons.installSpoonFromZipURL
+function _installSpoonFromZipURLgetCallback(urlparts, status, body, headers)
+   if status < 0 then
+      log.ef("Error downloading %s, error: %s", urlparts.absoluteURL, body)
+      return nil
+   else
+      -- Write the zip file to disk in a temporary directory
+      local tmpdir=_x("/usr/bin/mktemp -d", "Error creating temporary directory to download new spoon.")
+      if tmpdir then
+         local outfile = string.format("%s/%s", tmpdir, urlparts.lastPathComponent)
+         local f=assert(io.open(outfile, "w"))
+         f:write(body)
+         f:close()
+
+         -- Check its contents - only one *.spoon directory should be in there
+         output = _x(string.format("/usr/bin/unzip -l %s '*.spoon/' | /usr/bin/awk '$NF ~ /\\.spoon\\/$/ { print $NF }' | /usr/bin/wc -l", outfile),
+                     "Error examining downloaded zip file %s, leaving it in place for your examination.", outfile)
+         if output then
+            if (tonumber(output) or 0) == 1 then
+               -- Uncompress the zip file
+               if _x(string.format("/usr/bin/unzip %s -d %s 2>&1", outfile, tmpdir),
+                     "Error uncompressing file %s, leaving it in place for your examination.", outfile) then
+                  -- And finally, install it using Hammerspoon itself
+                  if _x(string.format("/usr/bin/open %s/*.spoon", tmpdir), "Error installing the spoon file %s/*.spoon", tmpdir) then
```

Comment:
@cmsj Is there a way to do this without the external `open` command? I.e. how to call Hammerspoon's `openFile:` Obj-C method from Lua.